pkg/logger/logger.go

```diff
@@ -24,18 +25,23 @@ var (
     /_/
  Observability + AIOps for Kubernetes
 `
-
 	DefaultLogLevel   = slog.LevelDebug
 	DefaultWriter     io.Writer
 	DefaultAddSource  = true
 	pluginGroupPrefix = "plugin"
 	NoRepeatInterval  = 3600 * time.Hour // arbitrarily long time to denote one-time sampling
+	logFs             afero.Fs
+	logFileName       = "opni-logs"
 	DefaultTimeFormat = "2006 Jan 02 15:04:05"
 	errKey            = "err"
 )
 
 var logSampler = &sampler{}
 
+func init() {
+	logFs = afero.NewMemMapFs()
```

Comment:
The plugin logs do not appear in the `debug agent-logs` output. Do the plugins share the same `logFs` here if they are different processes?

Reply:
No, they will be separate. The plugins should be sending their logs to the host process though, maybe that needs some refactoring.

Reply:
Although I wasn't able to stream the logs over `SyncStderr`, the `Logger` in the `ClientConfig` can forward the log messages to a writer, where I can log them with a logger on the host side.
```
Logger: hclog.New(&hclog.LoggerOptions{
			Level:  hclog.Debug,
			Output: logger.PluginFileWriter,
		}),
```
Go-plugin can't directly forward protobuf message bytes unfortunately, as it's hardcoded to cast all non-hclog output to a `string`: https://github.com/hashicorp/go-plugin/blob/017b758bf4d495212a55db3de61b2d95ab104e53/client.go#L1155

Reply:
The forwarded logs look like this. What do you think of the formatting?
<img width="1158" alt="image" src="https://github.com/rancher/opni/assets/84587295/c94e083b-7c1c-477a-9ea8-9d33961d95e0">

Reply:
Log streaming is working with the forked go-plugin patch in my k3s cluster. I replaced the hclog forwarding logger with `io.Copy(c.config.Stderr, r)`. I'm working on fixing the test env output because the plugins are logging the gprc bytes to stderr